### PR TITLE
新增离线玩家NBT读写API和完善PlayerAteEvent

### LINKED ISSUE #305: 希望可以添加修改离线玩家NBT的API
### Is your feature request related to a problem? Please describe. 您认为现在存在什么问题?

like this:
let offlineNBT = mc.getOfflinePlayerNBT(uuid/xuid/realname);
mc.setOfflinePlayerNBT(uuid/xuid/realname,nbt)

### Describe the solution you'd like 您认为还缺少什么?

_No response_

### PR DESCRIPTION
**Pull request by KobeBryant114514**
## Description
新增离线玩家NBT读写API  #305 
完善PlayerAteEvent并为LLSE添加该事件  #895 

## Issues fixed by this PR
新增了离线玩家NBT读写API
LIAPI static bool setPlayerNbt(mce::UUID const& uuid, CompoundTag& nbt);
LIAPI static bool setPlayerNbtTags(mce::UUID const& uuid, CompoundTag& data, vector<string> tags);
LIAPI static bool deletePlayerNbt(mce::UUID const& uuid);
LIAPI static std::unique_ptr<CompoundTag> getPlayerNbt(mce::UUID const& uuid);
issue #305

为LLSE添加PlayerAteEvent并改为TInstanceHook
issue #895

## Type of changes

- [x] Bug fix
- [x] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://docs.litebds.com/#/zh_CN/Maintenance/README)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: LiteLoader/src/llapi/mc/PlayerAPI.cpp [hunk 1]
```diff
@@ -935,6 +950,175 @@ bool Player::sendCustomFormPacket(const std::string& data, std::function<void(st
     });
 }
 
+inline class Player* getOfflinePlayer(class mce::UUID const& a0) {
+    class Player* (Level:: * rv)(class mce::UUID const&);
+    *((void**)&rv) = dlsym("?getPlayer@Level@@UEBAPEAVPlayer@@AEBVUUID@mce@@@Z");
```
Comment (Jaciezyt) [inline LiteLoader/src/llapi/mc/PlayerAPI.cpp:947]: ~~bdx时代的写法~~
建议使用
https://github.com/LiteLDev/LiteLoaderBDS/blob/a5eaf6290b0f3836075098148e2a5131746c4f25/LiteLoader/include/llapi/mc/Level.hpp#L1130

File: LiteLoader/src/llapi/mc/PlayerAPI.cpp [hunk 2]
```diff
@@ -935,6 +950,175 @@ bool Player::sendCustomFormPacket(const std::string& data, std::function<void(st
...
+
+std::string getServerId(mce::UUID const& uuid) {
+    auto tag = getPlayerIdsTag(uuid);
+    if (!tag){
+        return "";
+    }
+    return tag->getString(PLAYER_KEY_SERVER_ID);
+}
+
+bool deletePlayerNbt(mce::UUID const& uuid){
```
Comment (Jaciezyt) [inline LiteLoader/src/llapi/mc/PlayerAPI.cpp]: style: {前要加一个空格
下同（建议使用IDE自带的clang-format工具跑一下

**Review by Jaciezyt — CHANGES_REQUESTED** [review]
另外请更新LLSE的文档，进行测试以确保可用性（如果已经测试过了当我没说

**Comment by Jaciezyt** [comment]
按#305 的要求，LLSE的API也需要，麻烦大佬看看罢

**Comment by KobeBryant114514** [comment]
PlayerAteEvent经测试没有任何问题。
离线NBT的API这部分的代码我再改改，LLSE的适配我会看看（我一般不用LLSE所以说写的时候一般都没给脚本引擎写）
等会晚上重新PR

**Comment by Jaciezyt** [comment]
Duplicate of #929

**Pull request closed without merging**